mender-auth/cli/actions.cpp

```diff
@@ -35,17 +36,54 @@ namespace path = mender::common::path;
 
 namespace ipc = mender::auth::ipc;
 
-DaemonAction::DaemonAction(unique_ptr<crypto::PrivateKey> &&private_key) :
-	private_key_(move(private_key)) {
+shared_ptr<MenderKeyStore> keystore_from_config(
+	const conf::MenderConfig &config, const string &passphrase) {
+	bool static_key = false;
+	string pem_file;
+	string ssl_engine;
+
+	if (config.https_client.key != "") {
+		pem_file = config.https_client.key;
+		ssl_engine = config.https_client.ssl_engine;
+		static_key = true;
+	}
+	if (config.security.auth_private_key != "") {
+		pem_file = config.security.auth_private_key;
+		ssl_engine = config.security.ssl_engine;
+		static_key = true;
+	}
+	if (config.https_client.key == "" && config.security.auth_private_key == "") {
```

Comment:
Oh my.. This is the pyramid we have to solve to get the right key from conf? :see_no_evil:

Reply:
We are now three that don't get the details of this logic :see_no_evil: I took it verbatim from the [legacy client](https://github.com/mendersoftware/mender/blob/1a86f02ab83795b136f7d0bc08f5f39701ea3b21/cli/commands.go#L129).

@merlin-northern can you please help us in the discussion?

Reply:
this is historical reason. client has only notion of one private key (as you know looking at `app.NewAuthManager`). but in reality there can be two keys: one for the https client (in the case of mTLS) and one that defaults to let's say `/data/mender/mender-agent.pem`. the logic is simple: if none is set, we use the one from the default file. if both `Security.AuthPrivateKey` and `HttpsClient.Key` are set the former takes precedence. as you [can see ](https://github.com/mendersoftware/mender/blob/master/cli/commands.go#L129C13-L129C28) this is purely for the hardware security and for those who may not want to present client certificate during TLS handshake, but still use the "`/data/mender/mender-agent.pem`" from the hardware.

as long as you are here I would straighten this little knot. the requirements:

1. as a user I want to be able to provide a key and a certificate for the TLS handshake
2. as a user I want to be able to provide "the main key" (`/data/mender/mender-agent.pem`)
3. in both cases as a user I need to be able to specify the keys by file path or pkcs#11 uri
4. in both cases as a user I need to be able to specify the ssl engine to use

Reply:
it has been a while, but this is what I remember without further investigations.

Reply:
I added a note in the code to look into this during [MEN-6668](https://northerntech.atlassian.net/browse/MEN-6668)

Reply:
thank you.